src/Symfony/Bundle/Resources/config/maker.xml

```diff
@@ -14,6 +14,16 @@
             <argument type="service" id="api_platform.metadata.resource.name_collection_factory" />
             <tag name="maker.command" />
         </service>
+
```

Comment:
We must also deprecate the existing services.

Reply:
I've added `<deprecated>` option in the service declaration, is that enough or should I use `trigger_deprecation()` in DataProvider / DataPersiter constructors?

Reply:
I think you should just remove these services.

Reply:
isn't it some kind of BC break? I know those are just dev tools but if somehow someone uses 2.7 and is not willing to upgrade to 3.0, shouldn't they still be able to create data provider and persisters with the maker?

Reply:
2.7 has not been released yet, if some people are using it anyway, they have to know that it can break anytime :wink:

Reply:
ok, I've just realized it was never released! I was thinking it was an old feature

this all makes sens now